TESTING.md

```diff
@@ -218,6 +207,43 @@ cargo make \
 cargo make watch-test
 ```
 
+**Note**: You can provision nodes with karpenter by specifying `karpenter-block-device-mappings` in `Test.toml`.
+To follow the generic mapping, use the following configuration:
+
+```toml
+[aws-k8s.configuration.karpenter]
+test-type = "quick"
+karpenter-device-mapping = [
+    {name = "/dev/xvda", volumeType = "gp3", volumeSize = 4, deleteOnTermination = true},
+    {name = "/dev/xvdb", volumeType = "gp3", volumeSize = 20, deleteOnTermination = true},
+]
```

Comment:
Summarizing some feedback from chatting about this - I'd prefer that we expose the Karpenter resource agent and the block device mapping as two orthogonal capabilities, for example:

```
[aws-k8s.configuration.karpenter]
test-type = "quick"
resource-agent-type = "karpenter"
block-device-mapping = [ ... ]

[aws-k8s.configuration.ec2]
test-type = "quick"
resource-agent-type = "ec2"
block-device-mapping = [ ... ]
```

For the purposes of this PR, since the EC2 resource agent doesn't support a block device mapping yet, we can just assert that we don't have that specific combination, and fix it later.

Overall the code looks good - this just seems like a relatively minor change that would encourage broader use of Karpenter in testing.